command/vault/helper/secret_writer.go

```diff
@@ -17,14 +17,14 @@ type SecretWriter struct {
 }
 
 // WriteSecret ...
-func (w SecretWriter) WriteSecret(secret *config.Secret) error {
+func (w SecretWriter) WriteSecret(secret *config.Secret, env string) error {
 	var path string
 
 	// @TODO Make a dedicated type for writing non-secrets !
 	if strings.HasPrefix(secret.Path, "/") {
 		path = strings.TrimLeft(secret.Path, "/")
 	} else if secret.Application != nil {
-		path = fmt.Sprintf("secret/%s/%s", secret.Application.Name, secret.Path)
+		path = fmt.Sprintf("secret/%s/%s/%s", env, secret.Application.Name, secret.Path)
```

Comment:
this is a breaking, unrelated change to your PR title, please open a new PR for discussion on this